### PR TITLE
chore: prevent case reference cross site scripting (APPLICS-829)

### PR DESCRIPTION
**Pull request by BenSurgisonPINS**
## Describe your changes

APPLICS-829- https://pins-ds.atlassian.net/browse/APPLICS-829

- Sanitize the caseRef received within the URL
- Added unit test for above

## Type of change 🧩

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [x] Other (please explain in the description section above)

## Checklist before requesting a review

- [x] I have performed a self-review of my own code
- [x] I have double checked this work does not include any hardcoded secrets or passwords
- [ ] I have made corresponding changes to the documentation
- [x] I have provided details on how I have tested my code
- [x] I have referenced the ticket number above
- [x] I have provided a description of the ticket
- [x] I have included unit tests to cover any testable code changes

### CODE REVIEW

File: packages/applications-service-api/src/controllers/timetables.js [hunk 1]
```diff
@@ -7,10 +7,15 @@ const ApiError = require('../error/apiError');
...
 module.exports = {
 	async getTimetables(req, res) {
-		const { caseRef } = req.params;
+		const caseRef = encodeURIComponent(req.params.caseRef);
 
-		logger.debug(`Retrieving timetables by case reference ${caseRef} ...`);
 		try {
+			logger.debug(`Retrieving timetables by case reference ${caseRef} ...`);
+
+			if (caseRef !== req.query.caseRef) {
```
Comment (beejjacobs) [inline packages/applications-service-api/src/controllers/timetables.js:15]: Is this meant to be `req.params.caseRef` ?

Comment (BenSurgisonPINS) [reply packages/applications-service-api/src/controllers/timetables.js:15]: No it's not.  It's the following: 	
	`const caseRef = encodeURIComponent(req.params.caseRef);`

This is one of the requirements for the cross site scripting recommendation highlighted within the ticket.

Comment (BenSurgisonPINS) [reply packages/applications-service-api/src/controllers/timetables.js:15]: On second thoughts, in the logger.debug it's probably ok for it not to be encoded.  I'll go with your suggestion.

Comment (BenSurgisonPINS) [reply packages/applications-service-api/src/controllers/timetables.js:15]: Done

**Review by HadrianHughesKC — APPROVED** [review]

**Pull request merged**